test/test_format.py

```diff
@@ -8,523 +8,736 @@
+            "System`InputForm": "Subscript[a, 4]",
+            "System`OutputForm": "Subscript[a, 4]",
+        },
+        "mathml": {
+            "System`StandardForm": "<msub><mi>a</mi> <mn>4</mn></msub>",
+            "System`TraditionalForm": "<msub><mi>a</mi> <mn>4</mn></msub>",
+            "System`InputForm": "<mrow><mi>Subscript</mi> <mo>[</mo> <mrow><mi>a</mi> <mtext>,&nbsp;</mtext> <mtext>4</mtext></mrow> <mo>]</mo></mrow>",
+            "System`OutputForm": "<mrow><mi>Subscript</mi> <mo>[</mo> <mrow><mi>a</mi> <mtext>,&nbsp;</mtext> <mtext>4</mtext></mrow> <mo>]</mo></mrow>",
+        },
+        "tex": {
+            "System`StandardForm": "a_4",
+            "System`TraditionalForm": "a_4",
+            "System`InputForm": "\\text{Subscript}\\left[a, 4\\right]",
+            "System`OutputForm": "\\text{Subscript}\\left[a, 4\\right]",
+        },
+    },
+    "Subsuperscript[a, p, q]": {
+        "msg": "SubsuperscriptBox",
+        "text": {
+            "System`StandardForm": "Subsuperscript[a, p, q]",
+            "System`TraditionalForm": "Subsuperscript[a, p, q]",
+            "System`InputForm": "Subsuperscript[a, p, q]",
+            "System`OutputForm": "Subsuperscript[a, p, q]",
+        },
+        "mathml": {
+            "System`StandardForm": "<msubsup><mi>a</mi> <mi>p</mi> <mi>q</mi></msubsup>",
+            "System`TraditionalForm": "<msubsup><mi>a</mi> <mi>p</mi> <mi>q</mi></msubsup>",
+            "System`InputForm": "<mrow><mi>Subsuperscript</mi> <mo>[</mo> <mrow><mi>a</mi> <mtext>,&nbsp;</mtext> <mi>p</mi> <mtext>,&nbsp;</mtext> <mi>q</mi></mrow> <mo>]</mo></mrow>",
+            "System`OutputForm": "<mrow><mi>Subsuperscript</mi> <mo>[</mo> <mrow><mi>a</mi> <mtext>,&nbsp;</mtext> <mi>p</mi> <mtext>,&nbsp;</mtext> <mi>q</mi></mrow> <mo>]</mo></mrow>",
+        },
+        "tex": {
+            "System`StandardForm": "a_p^q",
+            "System`TraditionalForm": "a_p^q",
+            "System`InputForm": "\\text{Subsuperscript}\\left[a, p, q\\right]",
+            "System`OutputForm": "\\text{Subsuperscript}\\left[a, p, q\\right]",
+        },
+    },
+    # Here I use Integrate to simplify the input.
+    "Integrate[F[x], {x, a, g[b]}]": {
+        "msg": "Non trivial SubsuperscriptBox",
+        "text": {
+            "System`StandardForm": "Subsuperscript[∫, a, g[b]]\u2062F[x]\u2062\uf74cx",
+            "System`TraditionalForm": "Subsuperscript[∫, a, g(b)]\u2062F(x)\u2062\uf74cx",
+            "System`InputForm": "Integrate[F[x], {x, a, g[b]}]",
+            "System`OutputForm": "Integrate[F[x], {x, a, g[b]}]",
+        },
+        "mathml": {
+            "System`StandardForm": '<mrow><msubsup><mo>∫</mo> <mi>a</mi> <mrow><mi>g</mi> <mo>[</mo> <mi>b</mi> <mo>]</mo></mrow></msubsup> <mo form="prefix" lspace="0" rspace="0.2em">\u2062</mo> <mrow><mi>F</mi> <mo>[</mo> <mi>x</mi> <mo>]</mo></mrow> <mo form="prefix" lspace="0" rspace="0.2em">\u2062</mo> <mrow><mtext>\uf74c</mtext> <mi>x</mi></mrow></mrow>',
```

Comment:
I get saddened about stuff that has lots of gibberish in it like setting rspace to 0.2em, but I don't have a good solution for this right now.

Reply:
I could mark these tests as "fragile" and put them in the set of tests that we allow fails.  The good part to have these tests is that we notice when an "unrelated" change, has an impact on how the formatting is processed. In the current state of things, this happens a lot...